pkgs/development/tools/java/jprofiler/default.nix

```diff
@@ -51,49 +51,32 @@ let
     categories = [ "Development" ];
   };
 
-  linux = stdenv.mkDerivation {
-    inherit pname version src desktopItems;
-
-    nativeBuildInputs = [ makeWrapper copyDesktopItems ];
+  dontStrip = true;
 
-    installPhase = ''
-      runHook preInstall
-      cp -r . $out
+  installPhase = ''
+    runHook preInstall
 
-      rm -f $out/bin/updater
-      rm -rf $out/bin/linux-ppc*
-      rm -rf $out/bin/linux-armhf
-      rm -rf $out/bin/linux-musl*
+    cp -r . $out
 
-      for f in $(find $out/bin -type f -executable); do
-        wrapProgram $f --set JAVA_HOME "${jdk.home}"
-      done
+    for f in $(find $out/bin $out/.install4j -type f -executable); do
```

Comment:
What is ``.install4j`` doing in ``$out/``?

Reply:
@SuperSandro2000 `.install4j` is https://www.ej-technologies.com/products/install4j/overview.html and it is used as part of jprofiler.

To be honest I can't answer to your question better than "this is a part of distributive"

Reply:
Shouldn't that be somewhere like $out/share/jprofiler/ instead?

Reply:
@SuperSandro2000 nope, and releative paths are hardcoded:
```sh
√ nixpkgs % ls -la /nix/store/kmn9s77swl907mhkbdch2ysxijprl6aa-jprofiler-13.0.2/
total 72
dr-xr-xr-x    15 root  wheel     480 Jan  1  1970 .
drwxrwxr-t  2297 root  nixbld  73504 Aug  1 23:46 ..
dr-xr-xr-x    48 root  wheel    1536 Jan  1  1970 .install4j
dr-xr-xr-x     3 root  wheel      96 Jan  1  1970 Applications
-r--r--r--     1 root  wheel    2301 Jan  1  1970 README.html
dr-xr-xr-x    12 root  wheel     384 Jan  1  1970 api
dr-xr-xr-x    30 root  wheel     960 Jan  1  1970 bin
dr-xr-xr-x     5 root  wheel     160 Jan  1  1970 config
dr-xr-xr-x     7 root  wheel     224 Jan  1  1970 demo
dr-xr-xr-x     8 root  wheel     256 Jan  1  1970 integrations
dr-xr-xr-x    60 root  wheel    1920 Jan  1  1970 lib
-r--r--r--     1 root  wheel   12615 Jan  1  1970 license.html
-r--r--r--     1 root  wheel   11414 Jan  1  1970 license.txt
-r--r--r--     1 root  wheel      24 Jan  1  1970 release
dr-xr-xr-x     4 root  wheel     128 Jan  1  1970 share
√ nixpkgs %  
```

Reply:
oh, no. So many none FHS paths. Can we move the entire program to somewhere else? Then the relative paths are also correct again

Reply:
@SuperSandro2000 it is quite difficult and requires some binary patching which brokes signatures on macOS... To be honest, I prefer to not do it without very good reason.

Reply:
We can't just move everything as is to somewhere else? Bummer

Reply:
@SuperSandro2000 I can move everything into `$out/opt/jprofiler` and that makes moving `$out/bin/jporfiler` into `$out/opt/jprofiler/bin/jprofiler`.

Reply:
That would be better. Then add a symlink or wrapper to $out/bin.

Generally packages that are used by end users should not fill / with all kinds of none FHS files.